sopht_mpi/utils/mpi_utils_3d.py

```diff
@@ -265,3 +265,98 @@ def exchange_finalise(self):
         Finalizing non-blocking exchange ghost data between neighbors.
         """
         MPI.Request.Waitall(self.comm_requests)
+
+
+class MPIFieldIOCommunicator3D:
+    """
+    Class exclusive for field communication across ranks, initialises data types
+    that will be used for scattering global fields and aggregating local fields.
+    Builds dtypes based on ghost_size (determined from stencil width of the
+    employed kernel). This class wont be seen by the user, rather based on field
+    metadata we determine the properties here.
+    """
+
+    def __init__(self, ghost_size, mpi_construct):
+        # Use ghost_size to define indices for inner cell (actual data without
+        # halo)
+        if ghost_size < 0:
+            raise ValueError("ghost size has to be >= 0")
```

Comment:
^same as above.

Reply:
Fixed in [ac994ec](https://github.com/fankiat/sopht-mpi/pull/7/commits/ac994ecb1d393a64e6da14a3a85142bcc15b74e7)